main.py

```diff
@@ -160,6 +160,43 @@ def string2timestamp(s):
 def year2timestamp(y):
     return time.mktime(datetime.date(y, 1, 1).timetuple())
 
+@app.route('/new-features', methods=['POST'])
+def updatebyemail():
+    emailaddress = str(request.json['address'])
+    fname = str(request.json['fname'])
+    lname = str(request.json['lname'])
+
+    if len(fname)>40:
+        return  jsonify(respo='First name to long')
+    if len(lname)>40:
+        return  jsonify(respo='Last name to long')
+    if len(emailaddress)>40:
+        return jsonify(respo='Email too long', emailaddress = emailaddress)
+
+        #con = sqlite3.connect('emailAddresses.db')
+        #cur = con.cursor()
+        #cur.execute("Create Table if not EXISTS Emails(id INTEGER PRIMARY KEY, email Text)")
+        #cur.execute("Insert into Emails (email) VALUES (?)", (emailaddress,))
+        #con.commit()
+    #newsubscriber = Subscriber()
+    user_exists = User.query.filter(and_(User.first_name == fname, User.last_name == lname, User.email == emailaddress))
+    if user_exists.count()==0:
+        user = User (email = emailaddress, first_name = fname, last_name = lname, subscriber=Subscriber())
+        db_session.add(user)
+        db_session.commit()
+        return jsonify(respo='Subscription saved', )  # back to a string to produce a proper response
+    else:
+        subscription_exists = Subscriber.query.filter(Subscriber.user_id == user_exists[0].id)
+        if subscription_exists.count() == 0:
```

Comment:
Common code should be in a method or under the same condition, e.g. by checking if either `user_exists.count() > 0` or `subscription_exists.count() > 0` and returning `Subscription already exists` in that case; and otherwise adding the new user.